subprojects/language-scala/src/main/java/org/gradle/api/internal/tasks/scala/ZincScalaCompiler.java

```diff
@@ -199,16 +197,8 @@ public EntryLookup(ScalaJavaJointCompileSpec spec) {
 
         @Override
         public DefinesClass definesClass(File classpathEntry) {
-            Optional<DefinesClass> dc = analysis(classpathEntry).map(a -> a instanceof Analysis ? (Analysis) a : null).map(a -> new AnalysisBakedDefineClass(a));
-            return dc.orElseGet(() -> {
-                return definesClassCache.get(classpathEntry, new Factory<DefinesClass>() {
-                    @Nullable
-                    @Override
-                    public DefinesClass create() {
-                        return Locate.definesClass(classpathEntry);
-                    }
-                });
-            });
+            Optional<DefinesClass> dc = analysis(classpathEntry).map(a -> a instanceof Analysis ? (Analysis) a : null).map(AnalysisBakedDefineClass::new);
+            return dc.orElseGet(() -> definesClassCache.get(classpathEntry, Locate::definesClass));
```

Comment:
Is there a reason to declare `dc` as a local variable here?

Reply:
Yes, I think this is used so you don't need to specify the generic type on `map` since Java can infer the target type. Without the local variable it would be:
```
        @Override
        public DefinesClass definesClass(File classpathEntry) {
            return analysis(classpathEntry)
                .map(a -> a instanceof Analysis ? (Analysis) a : null)
                .<DefinesClass>map(AnalysisBakedDefineClass::new)
                .orElseGet(() -> definesClassCache.get(classpathEntry, Locate::definesClass));
        }
```

I don't feel strongly to go either way.

Reply:
`.map()` should be on a new line anyway.

I think the code you showed as an alternative is a significantly easier to read than what we have now.

Reply:
Changed to my suggestion in https://github.com/gradle/gradle/pull/15067.